### ISSUE TITLE
[explicit-module-boundary-types] Don't warn on user-defined type guard

### ISSUE THREAD

**Issue by lilred** [typescript-eslint/typescript-eslint#3161]
- [x] I have tried restarting my IDE and the issue persists.
- [ ] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-module-boundary-types": ["error"]
  }
}
```

```TS
export const isNumber = (v: any): v is number => {
  return typeof v === 'number';
}
```

**Expected Result**

No warning or error.

**Actual Result**

`Argument 'v' should be typed with a non-any type.`

**Additional Info**

N/A

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.0.1` |
| `@typescript-eslint/parser`        | `4.9.1` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `7.15.0` |
| `node`                             | `12.x` |

**Comment by bradzacher** [comment]
Nope!
Use `unknown` instead.
